# expressed-variant-impact

# Contributors
Brian Haas (bhaas@broadinstitute.org)

Olaitan I. Awe (laitanawe@gmail.com)

Meghana Pagadala (mpagadal@ucsd.edu)

Kym Pagel (kpagel1442@gmail.com)


# Intro statement


# Why Trinity CTAT?

The CTAT-Mutation pipeline (https://github.com/NCIP/ctat-mutations/wiki) makes it easy to discover variants from RNA-seq data, and requires only the RNA-seq reads as input. The pipeline also annotates variants, including the RADAR and RediPortal databases for identifying likely RNA-editing events, dbSNP and gnomAD for annotating common variants, COSMIC to highlight known cancer mutations, and OpenCRAVAT to annotate and prioritize variants according to likely biological impact and relevance to cancer.

The CTAT-Mutations Pipeline integrates GATK Best Practices along with downstream steps to annotate and filter variants, and to additionally prioritize variants that may be relevant to cancer biology. 


# How does this pipeline work?

![Trinity CTAT pipeline](https://github.com/NCIP/ctat-mutations/wiki/imgs/workflow.png) 

[Workflow](img/ctat_workflow.png)


# How to use the pipeline


## Input


## Output

The output from the CTAT-Mutations pipeline includes variant vcf files, summary tab-delimited reports, and interactive visualizations. The outputs include the following:

- variants.HC_init.wAnnot.vcf.gz contains HaplotypeCaller variant calls fully annotated. 
- A cancer.vcf and corresponding simpler summary cancer.tab file, contain a set of prioritized cancer-relevant variants detected in the sample. 
- Additional outputs that are generated by the different stages of the CTAT-Mutations pipeline, as others are likely to be of interest as well for exploring RNA-editing or common variants. 
- There is also an html page output (based on igv-reports), which allows for dynamic navigation of the identified cancer variants and the read evidence supporting their identification. 


## Annotations included by default

- Known or predicted RNA-editing site
- If the variant has entries in OMIM/OMIA or Clinical PubMed
- dbSNP ID and Variant Allele Origin (VAO, 0-unspecified 1-Germline 2-Somatic 3-Both)
- gnomAD allele frequency
- VAF
- Repeat family from UCSC Genome Browser Repeatmasker Annotations
- ADjacency to homopolymer sequence or splice sites
- Entropy around the variant
- COSMIC annotations, including somatic status
- HUGO Gene impacted and canonical transcript identifier
- Variant impact (sequence ontology)
- HGVS notation
- Number of samples that contain this variant
- CHASMplus driver status prediction
- ClinVar annotations
- Link to protein visualization
- Variant pathogenicity prediction by VEST and FATHMM

# Testing


# Installation
