# expressed-variant-impact

# Contributors

# Intro statement

# Why Trinity?

# How does it work?

# How to use

# Quickstart

## Input

## Output

# Annotations included by default
Hereâ€™s a list of the annotations in the output file

- If the variant has entries in OMIM/OMIA
- Variant is Precious(Clinical Pubmed Cited)
- dbSNP ID and Variant Allele Origin (VAO, 0-unspecified 1-Germline 2-Somatic 3-Both)
- gnomAD allele frequency
- Known or predicted RNA-editing site
- VAF
- Repeat family from UCSC Genome Browser Repeatmasker Annotations
- Homopolymer adjacency
- Entropy around the variant
- Splice adjacency
- FATHMM pathogenicity prediction
- COSMIC annotations, including somatic status
- Gene impacted
- Canonical transcript name
- Variant impact (sequence ontology)
- HGVS notation
- Number of samples that contain this variant
- CHASMplus driver status prediction
- ClinVar annotations
- Link to protein visualization
- Variant pathogenicity prediction by VEST and FATHMM

# Testing

# Installation

## Install Nim

## Install Nim

## Install Nim

Resources
- Trinity CTAT https://github.com/NCIP/ctat-mutations/wiki 
